src/lib.rs

```diff
@@ -1,22 +1,28 @@
-#![allow(non_upper_case_globals,non_snake_case)]
+// TODO: remove allow(private_in_public), and fix compile errors
+#![allow(non_upper_case_globals,non_snake_case,private_in_public)]
```

Comment:
I had to add this allow statement to get an easy fix for https://github.com/rust-lang/rust/issues/34537 because I didn't want to muddle this pr with a comprehensive fix.